app-modules/service-management/tests/ServiceRequest/CreateServiceRequestTest.php

```diff
@@ -464,11 +467,93 @@
+    User::factory()->licensed(LicenseType::cases())->create();
+    $team = Team::factory()->create();
+
+    $serviceRequestTypesWithManager = ServiceRequestType::factory()
+        ->hasAttached(
+            factory: $team,
+            relationship: 'managers'
+        )
+        ->state([
+            'assignment_type' => ServiceRequestTypeAssignmentTypes::RoundRobin,
+        ])
+        ->create();
+
+    $request = collect(CreateServiceRequestRequestFactory::new()->create([
+        'priority_id' => ServiceRequestPriority::factory()->create([
+            'type_id' => $serviceRequestTypesWithManager->getKey(),
+        ])->getKey(),
+    ]));
+    $users = User::orderBy('name')->orderBy('id')->get();
+
+    foreach ($users as $user) {
+        $user->grantLicense(LicenseType::RecruitmentCrm);
+        $user->teams()->attach($team);
+        $user->givePermissionTo('service_request.view-any');
+        $user->givePermissionTo('service_request.create');
+        $user->refresh();
+        actingAs($user);
+
+        livewire(CreateServiceRequest::class)
+            ->assertSuccessful()
+            ->fillForm($request->toArray())
+            ->call('create');
+
+        $serviceRequestDataObject = new ServiceRequestDataObject(
+            division_id: $request['division_id'],
+            status_id: $request['status_id'],
+            type_id: $request['type_id'],
+            priority_id: $request['priority_id'],
+            title: $request['title'],
+            close_details: $request['close_details'],
+            res_details: $request['res_details'],
+            respondent_type: $request['respondent_type'],
+            respondent_id: $request['respondent_id'],
+        );
+
+        app(CreateServiceRequestAction::class)->execute($serviceRequestDataObject);
+        $getServiceRequestType = ServiceRequestType::where('assignment_type', 'round-robin')->first();
+        expect($getServiceRequestType->round_robin_last_assigned_id)->ToBe($user->getKey());
+    }
+
+    livewire(CreateServiceRequest::class)
+        ->assertSuccessful()
+        ->fillForm($request->toArray())
+        ->call('create');
+
+    $serviceRequestDataObject = new ServiceRequestDataObject(
+        division_id: $request['division_id'],
+        status_id: $request['status_id'],
+        type_id: $request['type_id'],
+        priority_id: $request['priority_id'],
+        title: $request['title'],
+        close_details: $request['close_details'],
+        res_details: $request['res_details'],
+        respondent_type: $request['respondent_type'],
+        respondent_id: $request['respondent_id'],
+    );
+
+    app(CreateServiceRequestAction::class)->execute($serviceRequestDataObject);
+    $getServiceRequestType = ServiceRequestType::where('assignment_type', 'round-robin')->first();
+    expect($getServiceRequestType->round_robin_last_assigned_id)->ToBe(User::orderBy('name')->orderBy('id')->first()->getKey());
```

Comment:
You could do like a while loop for the total amount of users plus one instead of this, hard coded.